fish-rust/src/common.rs

```diff
@@ -1196,34 +1195,31 @@ pub fn should_suppress_stderr_for_tests() -> bool {
     unsafe { !PROGRAM_NAME.is_empty() && *PROGRAM_NAME != TESTS_PROGRAM_NAME }
 }
 
-fn assert_is_main_thread() {
-    assert!(is_main_thread() || THREAD_ASSERTS_CFG_FOR_TESTING.load());
+#[deprecated(note = "Use threads::assert_is_main_thread() instead")]
+pub fn assert_is_main_thread() {
+    crate::threads::assert_is_main_thread()
 }
 
-fn assert_is_background_thread() {
-    assert!(!is_main_thread() || THREAD_ASSERTS_CFG_FOR_TESTING.load());
+#[deprecated(note = "Use threads::assert_is_background_thread() instead")]
+pub fn assert_is_background_thread() {
+    crate::threads::assert_is_background_thread()
 }
 
-static THREAD_ASSERTS_CFG_FOR_TESTING: RelaxedAtomicBool = RelaxedAtomicBool::new(false);
-
-thread_local! {
-    static TL_TID: RefCell<u64> = RefCell::new(0);
-}
-
-static S_LAST_THREAD_ID: AtomicU64 = AtomicU64::new(0);
-fn next_thread_id() -> u64 {
-    // Note 0 is an invalid thread id.
-    // Note fetch_add is a CAS which returns the value *before* the modification.
-    1 + S_LAST_THREAD_ID.fetch_add(1, Ordering::Relaxed)
-}
+/// Converts the native rust [`ThreadId`](std::thread::ThreadId) to a `u64` for legacy interop
+/// purposes - this is the same type used by `is_main_thread()` and co.
+///
+/// Prefer to use the native type where possible.
+#[deprecated(note = "Use std::thread::current().id() instead")]
+pub fn thread_id() -> u64 {
```

Comment:
this one we can probably remove right away since there are few callers

Reply:
Yeah, I *think* it was mainly used by the other common.cpp helper fns themselves.